### ISSUE TITLE
Add option to disable random seed sharding and syntonization for DistributedReadingService

### ISSUE THREAD

**Issue by ejguan** [pytorch/data#1048]
### 🚀 The feature

This is a customer request for inference, no synchronization is needed because no `all_reduce` is needed. And, normally inference doesn't require shuffling and requires high throughput. Sharing random seed becomes redundant in this scenario.

We might want to add two more arguments to enable/disable synchronization.

### Motivation, pitch

Provide better performance for inference using DataPipe + DataLoader2

### Alternatives

Add a special ReadingService for distributed inference.
